# Research

Understanding Stdev in excel:https://www.youtube.com/watch?v=NdIVwuwfMIg&ab_channel=DesigneLearningTutorials <br />
http://mathbitsnotebook.com/Algebra1/StatisticsData/STPopSample.html <br />
Stdev.p is used to find the standard deviation of all the data acquired. The p stands for population. In excel, you run the formula while selecting all data cells in the spreadsheet to find the result.
Stdev.s is used to calculate the standard deviation of a sample of the data required. In excel you choose a select number of data cells and run the formula to find the standard deviation of those selected cells.

# Code

In [2]:
import numpy as np
#data is random results of a test in % format - 20 in total
data = ([75,22,82,48,51,92,72,64,38,57,62,79,88,28,53,83,75,66,43,52,78,91,85,34,58])
print("standard deviation sample result")
#formula for sample - np.sqrt(np.sum((x - np.mean(x))**2)/len(x))
np.sqrt(np.sum((data - np.mean(data))**2)/len(data))

standard deviation sample result


19.777724843874232

# References

In [3]:
import numpy as np
#data is random results of a test in % format - 20 in total
data = ([75,22,82,48,51,92,72,64,38,57,62,79,88,28,53,83,75,66,43,52,78,91,85,34,58])
print("standard deviation population result")
np.sqrt(np.sum((data - np.mean(data))**2)/len(data)-1)

standard deviation population result


19.75242769889312

# Analysis

The results of both tests gives the user a basis of where the standard deviation of the tests is based. Stdev.p = 19.78(to 2 decimal places) while stdev.s = 17.75(to two decimal places). The difference to two decimal places is 0.03. This gives the user an understanding that the data is mainly between 20% of the mean.
Using Stdev.s gives the user a better understanding of what the standard deviaton is because it will offer a greater fluctuation of results due it taking only select results that are randomly choosen from the data provided.When the data provided gets larger, the population result can change rapidly because alot of people could score around the same mark.This will skew the data and give out a result that may not accurately reflect the data provided to give the best outcome. Due to this using the sample standard deviation formula allows the data to be randomly choosen and give a better outcome because the data provided will be of a greater variance. 
With the data provided in the above code size was not 25 but 2500 and 5-% of the results are between 45-55, this will cause the standard dev population to be a much lower number than the number calculated in this file.